rustls/src/hash_hs.rs

```diff
@@ -15,10 +15,7 @@ use std::mem;
 /// This is disabled in cases where client auth is not possible.
 pub struct HandshakeHash {
     /// None before we know what hash function we're using
-    alg: Option<&'static digest::Algorithm>,
-
-    /// None before we know what hash function we're using
-    ctx: Option<digest::Context>,
+    ctx: Option<(&'static digest::Algorithm, digest::Context)>,
```

Comment:
`digest::Context::algorithm()` can be used to get the digest algorithm, so the first part of the tuple isn't needed and is bad because it makes us worry that it is inconsistent. We should remove the first part of the tuple and just leave the `Context`.